[RFC] Stored camera positions

mugmuggy opened the pull request:
When referring to the table here in [programming ideas](https://github.com/CorsixTH/CorsixTH/wiki/Programming-Ideas#key-bindings) I noted the only unimplemented original feature was having stored camera positions that you can recall with a key combo. Alt 0-9 to save a position and Ctrl 0-9 to restore them. Currently I don't have this changing the zoom level, so when zoomed out, the movement is limited but if zoomed in a bit, you can navigate to different part of the maps quickly.

I included the fix to address room queue dialog mood icons showing queue icons when dialog is interacted with - #789 - changes limited to onCursorWorldPositionChange.

TheCycoONE commented:
The hit test fix for 789 should be in a separate commit (I don't mind it in the same PR)

I don't understand the `--register` and `--deregister` comments where you are doing the key binding.

Have you tried with and without including the zoom level to see how it felt? I'm curious which is the better experience.

mugmuggy commented:
Sorry comments are probably a bit confusing, as I refactored and changed my mind a couple of times on the functionality, I'll fix that up with the commit being fixed up. Maybe I should have created an issue for discussion. I originally had one function, which I was going to call with true, false or nil parameter to set/recall/unset - unset is overkill.

Without the zoom, you could have 2 or more camera positions that are close and when you zoom right out it won't move, as the camera is limited, however with zoom, you can store a view that is zoomed all the way out, but just in typing that, sounds like a useful hotkey, a bit like transparent walls.

But with zoom feels better to me, but I haven't played much with the feature, as the intent really is to have a particular view that you can go back to easily.

TheCycoONE commented:
@mugmuggy this is still marked as WIP, what is the state of this PR?

mugmuggy commented:
It functionally works with the zoom, and it plays ok as it is so it is 'ready'. I'll just check that nothing is broken with a rebase and can push another commit and fix the conflict up too if you like.

TheCycoONE commented:
Sounds great. First feature for the next release!

TheCycoONE commented:
Bumped the SAVEGAME_VERSION and merged.
